src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  for (const key of toFormat) {
+    if (dateSet[key]) {
+      newDate.push(dateSet[key]);
+    } else if (key === 'YY') {
+      newDate.push(dateSet.YYYY.slice(2));
+    } else if (key === 'YYYY') {
+      if (dateSet.YY < 30) {
```

Comment:
The condition `if (dateSet.YY < 30)` is not safe since `dateSet.YY` is a string. You should convert it to a number before comparing it to `30` to ensure correct behavior.